ignore/src/types.rs

```diff
@@ -149,7 +149,7 @@ const DEFAULT_TYPES: &'static [(&'static str, &'static [&'static str])] = &[
     ("groovy", &["*.groovy", "*.gradle"]),
     ("h", &["*.h", "*.hpp"]),
     ("hbs", &["*.hbs"]),
-    ("haskell", &["*.hs", "*.lhs"]),
+    ("haskell", &["*.hs", "*.lhs", "*.cpphs", "*.c2hs", "*.hsc"]),
```

Comment:
Same question as for `cats` I think. When and where are these file extensions used?

Reply:
`.hsc` is for the `hsc2hs` preprocessor. You can see an example here: https://github.com/haskell-foundation/foundation/blob/master/foundation/Foundation/Foreign/MemoryMap/Posix.hsc

`.cpphs` is just for the `cpphs` preprocessor, e.g. here: https://github.com/vmchale/atspkg/blob/master/ats-pkg/internal/Quaalude.cpphs

`c2hs` is another pre-processor, and there is a nice example here: https://github.com/haskell/c2hs/blob/master/examples/libghttpHS/Ghttp.chs

Reply:
All righty, I buy it, thanks for showing me those!

Reply:
No problem!